### PR TITLE
Upgrade from Chromium 121 to Chromium 122

### PR DESCRIPTION
**Pull request by emerick**
Resolves https://github.com/brave/brave-browser/issues/35200

## Submitter Checklist:

- [ ] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [ ] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [ ] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

File: chromium_src/chrome/browser/ui/webui/help/version_updater_mac.h [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+  StatusCallback status_callback_;
+
+  // Callback used to show or hide the promote UI elements.
+  PromoteCallback promote_callback_;
+
+  // The visible state of the promote button.
+  bool show_promote_button_;
+
+  // The observer that will receive Keystone status updates.
+  KeystoneObserver* __strong keystone_observer_;
```
[the remark this answers is not in the document]
Comment (emerick) [reply chromium_src/chrome/browser/ui/webui/help/version_updater_mac.h:65]: FYI: This file was upstream code that was removed in cr122 and reinstated by us to support the Keystone integration points needed by our installer.

**Review by zenparsing — APPROVED** [review]
Rewards 👍

**Review by fmarier — APPROVED** [review]
`package.json` and Tor changes look good.

Licensing changes will be gone in the next rebase now that https://github.com/brave/brave-core/pull/21867 has been merged.

File: android/javatests/org/chromium/chrome/browser/safety_check/BraveSafetyCheckSettingsFragmentTest.java [hunk 2]
```diff
@@ -23,11 +23,12 @@
 // appeared in this section .
 @RunWith(ChromeJUnit4ClassRunner.class)
 public class BraveSafetyCheckSettingsFragmentTest {
-    private static final String PASSWORDS = "passwords";
+    private static final String PASSWORDS_LOCAL = "passwords_local";
```
[the remark this answers is not in the document]
Comment (fmarier) [reply android/javatests/org/chromium/chrome/browser/safety_check/BraveSafetyCheckSettingsFragmentTest.java:26]: False positive

File: android/javatests/org/chromium/chrome/browser/safety_check/BraveSafetyCheckSettingsFragmentTest.java [hunk 3]
```diff
@@ -23,11 +23,12 @@
...
+    private static final String PASSWORDS_ACCOUNT = "passwords_account";
```
[the remark this answers is not in the document]
Comment (fmarier) [reply android/javatests/org/chromium/chrome/browser/safety_check/BraveSafetyCheckSettingsFragmentTest.java:27]: false positive

File: patches/build-rust-cargo_crate.gni.patch [hunk 4]
```diff
@@ -10,3 +10,11 @@ index 1c86569e45153d9bfad207c65ca728373adb9d84..e523e46a017da8f2efcc858af6280807
  
    # cargo_crate() should set library_configs, executable_configs,
    # proc_macro_configs. Not configs.
+@@ -236,6 +237,7 @@ template("cargo_crate") {
+     testonly = _testonly
+     if (defined(invoker.visibility)) {
+       visibility = invoker.visibility
++      visibility += [ "//brave/third_party/*", "//brave/components/*",  ]
```
Comment (rillian) [inline patches/build-rust-cargo_crate.gni.patch:17]: Seems like a reasonable response, although I'm unclear why this is only appropriate inside the conditional? Is it because `invoker.visibility` is only defined if we're inheriting, and we assume a correct local declaration?

File: third_party/rust/indexmap/v1/BUILD.gn [hunk 5]
```diff
@@ -14,7 +14,6 @@ cargo_crate("lib") {
 
   # Only for usage from third-party crates. Add the crate to
   # third_party.toml to use it from first-party code.
-  visibility = [ "//third_party/rust/*" ]
```
Comment (rillian) [inline third_party/rust/indexmap/v1/BUILD.gn:17]: This should be
```gn
visibility = [ "//brave/third_party/rust/*" ]
```
list all the sibling BUILD.gn, rather than left public.

File: android/java/org/chromium/chrome/browser/ntp_background_images/RewardsBottomSheetDialogFragment.java
Comment (mkarolin) [inline android/java/org/chromium/chrome/browser/ntp_background_images/RewardsBottomSheetDialogFragment.java:1]: Looks like all the *.java files belong to a different commit (not to https://github.com/brave/brave-core/pull/21524/commits/d7b89fd211ab4337546c52c8a4cab905ffa256da).

File: chromium_src/content/public/common/content_features.cc [hunk 6]
```diff
@@ -23,7 +23,6 @@ OVERRIDE_FEATURE_DEFAULT_STATES({{
     {kDigitalGoodsApi, base::FEATURE_DISABLED_BY_DEFAULT},
     {kDIPS, base::FEATURE_DISABLED_BY_DEFAULT},
     {kFedCm, base::FEATURE_DISABLED_BY_DEFAULT},
-    {kFirstPartySets, base::FEATURE_DISABLED_BY_DEFAULT},
```
Comment (mkarolin) [inline chromium_src/content/public/common/content_features.cc:26]: Seems like we'd want to have the affected code disabled in another way then.

Comment (emerick) [reply chromium_src/content/public/common/content_features.cc:26]: Yeah, that's not right. @samartnik do you happen to know what commit those files belong with?

File: android/java/org/chromium/chrome/browser/ntp_background_images/RewardsBottomSheetDialogFragment.java
Comment (emerick) [reply android/java/org/chromium/chrome/browser/ntp_background_images/RewardsBottomSheetDialogFragment.java:1]: Yeah, that's not right. @samartnik do you happen to know what commit those files belong with?

File: chromium_src/content/public/common/content_features.cc (hunk 6 above)
Comment (emerick) [reply chromium_src/content/public/common/content_features.cc:26]: cc: @ShivanKaul

Comment (emerick) [reply chromium_src/content/public/common/content_features.cc:26]: @ShivanKaul wrote:

> The new name for First Party Sets is Related Website Sets. I think as long as we make the following function return false, it will be equivalent to the flag being off: [PrivacySandboxSettings::AreRelatedWebsiteSetsEnabled()](https://source.chromium.org/chromium/chromium/src/+/main:components/privacy_sandbox/privacy_sandbox_settings.h;l=332;drc=805c9ee2d426bd2ef36ec6128553bed917f1b5e4;bpv=1;bpt=1)

We provide a stub implementation of `PrivacySandboxSettings` that returns `false` from that function here:
https://github.com/brave/brave-core/blob/master/components/privacy_sandbox/brave_privacy_sandbox_settings.cc#L224-L226

File: android/java/org/chromium/chrome/browser/ntp_background_images/RewardsBottomSheetDialogFragment.java
Comment (samartnik) [reply android/java/org/chromium/chrome/browser/ntp_background_images/RewardsBottomSheetDialogFragment.java:1]: I don't remember these changes, it looks like just formatting is done in them

Comment (samartnik) [reply android/java/org/chromium/chrome/browser/ntp_background_images/RewardsBottomSheetDialogFragment.java:1]: So yeah, it's just format fixes for this change https://github.com/brave/brave-core/commit/7750bfc08170dbb5e7b94d6329eb95ec8d997fe4

File: browser/themes/brave_dark_mode_utils.cc [hunk 7]
```diff
@@ -28,11 +28,6 @@ namespace {
 bool g_is_test_ = false;
 bool g_system_dark_mode_enabled_in_test_ = false;
 
-void ClearBraveDarkModeProfilePrefs(PrefService* prefs) {
-  prefs->ClearPref(kBraveThemeType);
```
Comment (simonhong) [inline browser/themes/brave_dark_mode_utils.cc:32]: Let's delete `kBraveThemeType` and `kUseOverriddenBraveThemeType` vars from `components/constants/pref_names.h` together. 👍🏼

**Review by simonhong — APPROVED** [review]
`browser/thems`, `browser/ui/views/infobar` and `browser/widevine` ++

File: third_party/rust/indexmap/v1/BUILD.gn (hunk 5 above)
Comment (emerick) [reply third_party/rust/indexmap/v1/BUILD.gn:17]: Fixed in https://github.com/brave/brave-core/pull/21524/commits/1f2b283553031b8db4f0b673243f7b2a79015a25.

File: android/java/org/chromium/chrome/browser/ntp_background_images/RewardsBottomSheetDialogFragment.java
Comment (emerick) [reply android/java/org/chromium/chrome/browser/ntp_background_images/RewardsBottomSheetDialogFragment.java:1]: Fixed via rebasing here https://github.com/brave/brave-core/pull/21524/commits/ad50367341686fb839296b71f6ce7fae01d1a9ad and here https://github.com/brave/brave-core/pull/21524/commits/5d06bf2d0bef9333c670cafb072134fcc5dbf737.

File: browser/themes/brave_dark_mode_utils.cc (hunk 7 above)
Comment (emerick) [reply browser/themes/brave_dark_mode_utils.cc:32]: Fixed in https://github.com/brave/brave-core/pull/21524/commits/1ac1432c6269c62ef9519d2a74448779b2f7c781.

File: patches/build-rust-cargo_crate.gni.patch (hunk 4 above)
Comment (cdesouza-chromium) [reply patches/build-rust-cargo_crate.gni.patch:17]: Tagging @bridiver because I had discussed with him about the whole visibility issue with rust crates, and he might have something in the works for this.

**Review by darkdh — APPROVED** [review]
tor, sync and wallet ++

**Review by mkarolin — APPROVED** [review]
`strings`++

**Review by mherrmann — APPROVED** [review]
I tested automatic updates on macOS via Keystone / Sparkle. Got a delta update on x64 for on-demand updates via `brave://settings/help`, and also a successful background update (i.e. with the browser running, but without going to `brave://settings/help`.)

++

File: browser/mac/BUILD.gn [hunk 8]
```diff
@@ -0,0 +1,26 @@
+# Copyright (c) 2024 The Brave Authors. All rights reserved.
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this file,
+# You can obtain one at https://mozilla.org/MPL/2.0/.
+
+check_includes = false
```
Comment (bsclifton) [inline browser/mac/BUILD.gn:6]: Did you check to see how bad things are with this removed? (ex: when running `npm run gn_check`). I know we try to avoid adding new instances of this. But maybe a follow up is a better place to address

Comment (emerick) [reply browser/mac/BUILD.gn:6]: That's a good point, I forgot I had to do that. Unfortunately, without that `gn_check` failed like this and I wasn't too sure how to get around that:

```
$ npm run gn_check Static --channel=nightly --target_arch=x64

> brave-core@1.64.51 gn_check
> node ./build/commands/scripts/commands.js gn_check Static

-----------------------------------------------------------------------------------------------------------------------
/Users/emerick/work/cr122/brave-browser/src
> gn check /Users/emerick/work/cr122/brave-browser/src/out/Static
ERROR at //brave/browser/mac/keystone_glue.mm:43:11: Include not allowed.
#include "brave/browser/mac/sparkle_glue.h"
          ^-------------------------------
It is not in any dependency of
  //brave/browser/mac:keystone_glue
The include file is in the target(s):
  //chrome/browser:browser
which should somehow be reachable.
```

Maybe @bridiver or @goodov have some advice here?

**Review by rillian — APPROVED** [review]
rust++

File: browser/mac/BUILD.gn (hunk 8 above)
Comment (bsclifton) [reply browser/mac/BUILD.gn:6]: Per DM, bad things happen with this removed! Thanks for helping share impact 😄 

Nothing to see here
homer

File: app/brave_strings.grd [hunk 9]
```diff
@@ -185,88 +185,88 @@ If you update this file, be sure also to update google_chrome_strings.grd. -->
     </if>
   </outputs>
   <translations>
-    <file path="resources/chromium_strings_af.xtb" lang="af" />
```
Comment (bsclifton) [inline app/brave_strings.grd:188]: ++

**Review by bsclifton — APPROVED** [review]
string reviewers and js_deps ++

**Review by AlexeyBarabash — APPROVED** [review]
sync++, basic sync functions for Android <=>  desktop work well

File: patches/build-rust-cargo_crate.gni.patch (hunk 4 above)
Comment (bridiver) [reply patches/build-rust-cargo_crate.gni.patch:17]: this is still needed, but shouldn't there be a removal of a bunch of visibility patches?

Comment (bridiver) [reply patches/build-rust-cargo_crate.gni.patch:17]: @rillian this is specifically to override when upstream has visibility set to  `//third_party/rust`

File: patches/chrome-browser-ui-omnibox-chrome_omnibox_client.h.patch [hunk 10]
```diff
@@ -0,0 +1,20 @@
...
+ #include "components/omnibox/browser/omnibox.mojom-shared.h"
+ #include "components/omnibox/browser/omnibox_client.h"
++#include "brave/components/omnibox/browser/brave_omnibox_client.h"
+ 
+ class Browser;
+ class GURL;
+ class LocationBar;
+ class Profile;
+ 
+-class ChromeOmniboxClient final : public OmniboxClient {
```
Comment (bridiver) [inline patches/chrome-browser-ui-omnibox-chrome_omnibox_client.h.patch:16]: I'm a bit confused on this, why does the original override break now? Do you possibly just need to add the include for omnibox_client.h in the override before the define?

File: android/java/org/chromium/base/Callbacks.java [hunk 11]
```diff
@@ -0,0 +1,248 @@
+/* Copyright (c) 2024 The Brave Authors. All rights reserved.
```
Comment (bridiver) [inline android/java/org/chromium/base/Callbacks.java:1]: are we going to fix our code in a follow-up?

File: chromium_src/chrome/renderer/url_loader_throttle_provider_impl.h [hunk 12]
```diff
@@ -0,0 +1,19 @@
+/* Copyright (c) 2024 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#ifndef BRAVE_CHROMIUM_SRC_CHROME_RENDERER_URL_LOADER_THROTTLE_PROVIDER_IMPL_H_
+#define BRAVE_CHROMIUM_SRC_CHROME_RENDERER_URL_LOADER_THROTTLE_PROVIDER_IMPL_H_
+
+#define BRAVE_URL_LOADER_THROTTLE_PROVIDER_IMPL_H_                   \
```
Comment (bridiver) [inline chromium_src/chrome/renderer/url_loader_throttle_provider_impl.h:9]: https://bravesoftware.slack.com/archives/CU5UXM3EC/p1705545054988689?thread_ts=1705420692.572819&cid=CU5UXM3EC

File: android/java/org/chromium/base/Callbacks.java (hunk 11 above)
Comment (samartnik) [reply android/java/org/chromium/base/Callbacks.java:1]: yes, this would be advisable, created a follow up issue https://github.com/brave/brave-browser/issues/35872

Comment (emerick) [reply android/java/org/chromium/base/Callbacks.java:1]: cc: @samartnik

File: components/content_settings/core/browser/brave_content_settings_pref_provider.cc [hunk 13]
```diff
@@ -843,7 +843,7 @@ void BravePrefProvider::NotifyChanges(
     bool incognito) {
   for (const auto& rule : rules) {
     Notify(rule->primary_pattern, rule->secondary_pattern,
-           ContentSettingsType::COOKIES);
+           ContentSettingsType::COOKIES, /*partition_key=*/nullptr);
```
Comment (bridiver) [inline components/content_settings/core/browser/brave_content_settings_pref_provider.cc:846]: isn't this important here? @goodov ?

File: patches/build-rust-cargo_crate.gni.patch (hunk 4 above)
Comment (rillian) [reply patches/build-rust-cargo_crate.gni.patch:17]: Makes sense, thanks.

File: chromium_src/components/content_settings/core/common/cookie_settings_base.cc [hunk 14]
```diff
@@ -292,13 +292,19 @@ CookieSettingsBase::GetCurrentCookieSettingWithBraveMetadata() {
 // were explicit. We use explicit setting to enable 1PES mode, but in this mode
 // we still want to block 3p frames as usual and not fallback to "allow
 // everything" path.
+#define BRAVE_COOKIE_SETTINGS_BASE_DECIDE_ACCESS                              \
```
Comment (bridiver) [inline chromium_src/components/content_settings/core/common/cookie_settings_base.cc:295]: seems like it would be better to rename `CookieSettingsBase` -> `CookieSettingsBaseChromiumImpl` and then create `CookieSettingsBase` that subclasses `CookieSettingsBaseChromiumImpl`?

Comment (bridiver) [reply chromium_src/components/content_settings/core/common/cookie_settings_base.cc:295]: I'm also a bit concerned about the complexity of this and not sure why it's so complex, but I guess that's a follow-up issue. Are there tests that cover all of this?

File: patches/third_party-blink-renderer-modules-storage-cached_storage_area.cc.patch [hunk 15]
```diff
@@ -0,0 +1,13 @@
+diff --git a/third_party/blink/renderer/modules/storage/cached_storage_area.cc b/third_party/blink/renderer/modules/storage/cached_storage_area.cc
+index cb7bf9e97a0138904c0d9f9522e335f4e493d79a..557cf4c91f832d56d13f3a7b7cb635cc8cd029fe 100644
+--- a/third_party/blink/renderer/modules/storage/cached_storage_area.cc
++++ b/third_party/blink/renderer/modules/storage/cached_storage_area.cc
+@@ -244,7 +244,7 @@ void CachedStorageArea::BindStorageArea(
+     remote_area_.Bind(std::move(new_area), task_runner);
+   } else if (storage_namespace_) {
+     storage_namespace_->BindStorageArea(
```
Comment (bridiver) [inline patches/third_party-blink-renderer-modules-storage-cached_storage_area.cc.patch:8]: `#define BindStorageArea(STORAGE_KEY, LOCAL_FRAME_TOKEN, RECEIVER) BindStorageArea(STORAGE_KEY, IsSessionStorage() ? storage_key_ : local_dom_window->GetEphemeralStorageKeyOrStorageKey(), local_dom_window->GetLocalFrameToken(), RECEIVER)

Comment (emerick) [reply patches/third_party-blink-renderer-modules-storage-cached_storage_area.cc.patch:8]: Can we do that? That's trying to replace this:

```
storage_namespace_->BindStorageArea(
    storage_key_, local_dom_window->GetLocalFrameToken(),
    remote_area_.BindNewPipeAndPassReceiver(task_runner));
```

but there are also two-parameter variants with that same function name in this scope, such as:

```
BindStorageArea(std::move(storage_area), local_dom_window);
```

Comment (bridiver) [reply patches/third_party-blink-renderer-modules-storage-cached_storage_area.cc.patch:8]: that shouldn't matter because it's defining a macro-like function with three parameters so it shouldn't match the two param version

Comment (bridiver) [reply patches/third_party-blink-renderer-modules-storage-cached_storage_area.cc.patch:8]: this is different than a regular macro matching `BindStorageArea` which would match both

Comment (emerick) [reply patches/third_party-blink-renderer-modules-storage-cached_storage_area.cc.patch:8]: Yeah, I definitely see what you're saying, but the compiler seems not to like it:

```
In file included from ../../brave/chromium_src/third_party/blink/renderer/modules/storage/cached_storage_area.cc:32:
../../..\src/third_party/blink/renderer/modules/storage/cached_storage_area.cc(199,60): error: too few arguments provided to function-like macro invocation
  199 |   BindStorageArea(std::move(storage_area), local_dom_window);
      |                                                            ^
../../brave/chromium_src/third_party/blink/renderer/modules/storage/cached_storage_area.cc(25,9): note: macro 'BindStorageArea' defined here
   25 | #define BindStorageArea(STORAGE_KEY, LOCAL_FRAME_TOKEN, RECEIVER)   \
      |         ^
```

It's possible I'm missing something, though. I've asked @cdesouza-chromium to take a look to double-check me there.

File: chromium_src/components/content_settings/core/common/cookie_settings_base.cc (hunk 14 above)
Comment (emerick) [reply chromium_src/components/content_settings/core/common/cookie_settings_base.cc:295]: I'll let @cdesouza-chromium respond regarding cleaning up the override, but yes we do have pretty comprehensive test coverage for this change with upstream tests (those are what led us to this solution).

File: patches/chrome-browser-ui-omnibox-chrome_omnibox_client.h.patch (hunk 10 above)
Comment (cdesouza-chromium) [reply patches/chrome-browser-ui-omnibox-chrome_omnibox_client.h.patch:16]: This change is still marked as WIP because I wanted to discuss it during review.

We have to issues here:

 * `ChromeOmniboxClient` is now tagged as `final`, and it would be UB to try to redefine `final` to nothing. We already ha
 * `virtual base::WeakPtr<OmniboxClient> AsWeakPtr() = 0;` has been introduced in the offending CL [1]. The current code on master relies on a substitution of `#define OmniboxClient BraveOmniboxClient` to correct the inheritance for  `ChromeOmniboxClient`, but in the process introducing a substitution in `base::WeakPtr<OmniboxClient> AsWeakPtr() override;`, which breaks the virtual inheritance.

One way to avoid the patching altogether might be doing away entirely with `BraveOmniboxClient`, which only seems to exist as a way to introduce `virtual bool IsAutocompleteEnabled() const;`

Comment (cdesouza-chromium) [reply patches/chrome-browser-ui-omnibox-chrome_omnibox_client.h.patch:9]: Simplified in 0e85a99ca65cd5a8a338aa5ba1c2107e48386564.

I have been unable to remove the patch entirely because of the final keyword. But I've done away with `BraveOmniboxClient`, which seemed to exist simply to insert `IsAutocompleteEnabled` in the inheritance chain. It seems to me now that we have a much simpler substitution scheme, by injecting `IsAutocompleteEnabled` directly into the base class `OmniboxClient`, and with that we now can avoid a dangerous downcast operation that was being done from `OmniboxClient` to `BraveOmniboxClient`.

File: chromium_src/components/content_settings/core/common/cookie_settings_base.cc (hunk 14 above)
Comment (cdesouza-chromium) [reply chromium_src/components/content_settings/core/common/cookie_settings_base.cc:295]: These overrides are changing significantly in `cr123`, and @goodov has in fact fixed some of of the fixes we did in `cr123`.

File: browser/mac/BUILD.gn (hunk 8 above)
Comment (emerick) [reply browser/mac/BUILD.gn:6]: That makes sense. I believe I tried that and it got a little complicated. I think given the timing with this release and the testing required for the updater, we'll have to address this one in a follow-up issue. I'll enter that issue now and will assign to myself.

Comment (emerick) [reply browser/mac/BUILD.gn:6]: https://github.com/brave/brave-browser/issues/35893

File: patches/third_party-blink-renderer-modules-storage-cached_storage_area.cc.patch (hunk 15 above)
Comment (emerick) [reply patches/third_party-blink-renderer-modules-storage-cached_storage_area.cc.patch:8]: I think we'll go with what we have for now, but happy to address this in a follow-up if there's some better way forward. @cdesouza-chromium ran this through Compiler Explorer, though, and it seems like it's not supported: https://godbolt.org/z/jbrYxvfsE

File: chromium_src/chrome/renderer/url_loader_throttle_provider_impl.h (hunk 12 above)
Comment (darkdh) [reply chromium_src/chrome/renderer/url_loader_throttle_provider_impl.h]: yep we should override `URLLoaderThrottleProviderImpl::Create` so we don't have to copy 
```
#if BUILDFLAG(ENABLE_EXTENSIONS)
  mojo::PendingRemote<safe_browsing::mojom::ExtensionWebRequestReporter>
      pending_extension_web_request_reporter;
  broker->GetInterface(
      pending_extension_web_request_reporter.InitWithNewPipeAndPassReceiver());
#endif  // BUILDFLAG(ENABLE_EXTENSIONS)
```
if we keep `BraveURLLoaderThrottleProviderImpl::Create`, stuffs can be easily broken during future chromium upgrades and hard to detect.

Comment (darkdh) [reply chromium_src/chrome/renderer/url_loader_throttle_provider_impl.h]: With that approach, we also need to override `URLLoaderThrottleProviderImpl::CreateThrottles` in order to insert our `tor::OnionDomainThrottle`

Comment (emerick) [reply chromium_src/chrome/renderer/url_loader_throttle_provider_impl.h]: OK, makes sense. Since we elminated the patch and given the current timing I'm going to open a follow-up issue to address this one.

Comment (emerick) [reply chromium_src/chrome/renderer/url_loader_throttle_provider_impl.h]: https://github.com/brave/brave-browser/issues/35897

File: patches/chrome-browser-ui-omnibox-chrome_omnibox_client.h.patch (hunk 10 above)
Comment (emerick) [reply patches/chrome-browser-ui-omnibox-chrome_omnibox_client.h.patch:16]: I had to revert this simplification @cdesouza-chromium, it was causing all of our browser tests to fail. Let's investigate simplifying this in the scope of cr123. I'll enter a tracking issue for us.

Comment (emerick) [reply patches/chrome-browser-ui-omnibox-chrome_omnibox_client.h.patch:16]: https://github.com/brave/brave-browser/issues/35904

File: patches/third_party-blink-renderer-modules-storage-cached_storage_area.cc.patch (hunk 15 above)
Comment (goodov) [reply patches/third_party-blink-renderer-modules-storage-cached_storage_area.cc.patch:8]: Everything here will be removed soon. Please don't waste time.

**Review by goodov — APPROVED** [review]

File: patches/chrome-browser-ui-omnibox-chrome_omnibox_client.h.patch (hunk 10 above)
Comment (bridiver) [reply patches/chrome-browser-ui-omnibox-chrome_omnibox_client.h.patch:16]: as mentioned in DM I'm not understanding while `final` is an issue here since we are not subclassing `ChromeOmniboxClient`, we're subclassing `OmniboxClient`

File: chromium_src/components/content_settings/core/common/cookie_settings_base.cc (hunk 14 above)
Comment (bridiver) [reply chromium_src/components/content_settings/core/common/cookie_settings_base.cc:295]: @goodov that makes sense for the complexity of the check, but still seems like the override could eliminate the need for patching here?

File: patches/third_party-blink-renderer-modules-storage-cached_storage_area.cc.patch (hunk 15 above)
Comment (bridiver) [reply patches/third_party-blink-renderer-modules-storage-cached_storage_area.cc.patch:8]: @emerick I suppose it makes sense that it just tries to match all instances of `BindStorageArea` and then fails on the one that doesn't have the correct number of params. We're kind of abusing the preprocessor here so I guess I shouldn't expect it to handle this the way we would like.

File: patches/chrome-browser-ui-omnibox-chrome_omnibox_client.h.patch (hunk 10 above)
Comment (bridiver) [reply patches/chrome-browser-ui-omnibox-chrome_omnibox_client.h.patch:16]: I'm not going to block on this, but lets discuss a follow-up

File: chromium_src/components/content_settings/core/common/cookie_settings_base.cc (hunk 14 above)
Comment (bridiver) [reply chromium_src/components/content_settings/core/common/cookie_settings_base.cc:295]: we can discuss if there should be a follow-up for this patch or not
